client/blocks/login/two-factor-authentication/two-factor-actions.jsx

```diff
@@ -0,0 +1,94 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import page from 'page';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import { localize } from 'i18n-calypso';
+import {
+	getTwoFactorUserId,
+	getTwoFactorAuthNonce,
+	isTwoFactorAuthTypeSupported,
+} from 'state/login/selectors';
+import { sendSmsCode } from 'state/login/actions';
+import { errorNotice, successNotice } from 'state/notices/actions';
+import { login } from 'lib/paths';
+
+class TwoFactorActions extends Component {
+	static propTypes = {
+		errorNotice: PropTypes.func.isRequired,
+		isAuthenticatorSupported: PropTypes.bool,
+		isSmsSupported: PropTypes.bool,
+		successNotice: PropTypes.func.isRequired,
+		twoFactorAuthType: PropTypes.string.isRequired,
+		twoStepNonce: PropTypes.string.isRequired,
+	};
+
+	sendSmsCode = ( event ) => {
+		event.preventDefault();
+
+		const { userId, twoStepNonce } = this.props;
+
+		this.props.sendSmsCode( userId, twoStepNonce ).then( () => {
+			page( login( { twoFactorAuthType: 'sms' } ) );
+		} ).catch( ( errorMesssage ) => {
+			this.props.errorNotice( errorMesssage );
+		} );
+	};
+
+	render() {
+		const {
+			isAuthenticatorSupported,
+			isPushSupported,
+			isSmsSupported,
+			translate,
+			twoFactorAuthType,
+		} = this.props;
+
+		return (
+			<Card className="two-factor-authentication__form-action is-compact">
+				<p>
+					{ translate( 'Or continue to your account using:' ) }
+				</p>
+
+				{ isSmsSupported && twoFactorAuthType !== 'sms' && (
+					<p>
+						<a href="#" onClick={ this.sendSmsCode }>{ translate( 'Code via text message' ) }</a>
+					</p>
+				) }
+
+				{ isAuthenticatorSupported && twoFactorAuthType !== 'authenticator' && (
+					<p>
+						<a href={ login( { twoFactorAuthType: 'authenticator' } ) }>{ translate( 'An Authenticator application' ) }</a>
+					</p>
+				) }
+
+				{ isPushSupported && twoFactorAuthType !== 'push' && (
+					<p>
+						<a href={ login( { twoFactorAuthType: 'push' } ) }>{ translate( 'The WordPress mobile app' ) }</a>
+					</p>
+				) }
+			</Card>
+		);
+	}
```

Comment:
We should not display that block when the user only has enabled SMS as two step authentication method because otherwise the list of actions is empty:

<img width="308" alt="screenshot" src="https://cloud.githubusercontent.com/assets/594356/26033594/9d470d60-38af-11e7-8255-7c5f8643637c.png">

For reference, here is an example of response returned by the login endpoint in that case:

```json
{
  "success": true,
  "data": {
    "user_id": 1234567890,
    "two_step_nonce": "blablabla",
    "two_step_supported_auth_types": ["sms"],
    "two_step_notification_sent": "sms"
  }
}
```